### ISSUE TITLE
Rocker-image build fails after experimenting within R instance in rootless Rocker-container

### ISSUE THREAD

**Issue by IlmariTamminen** [containers/podman#18916]
### Issue Description

Hi.

I was experimenting with the renv R package in an R instance contained by a rootless rocker/r-ver:4.2.1 container. For some reason, I am not able to build the Rocker images anymore, although the container is already shutdown. I used the Podman 4.5.0 (with podman-machine-default) in Windows 10 Enterprise with WSL2 installed. It is possible that I have done something silly, or the issue is already fixed in the newer Podman. However, what is the explanation for the observed behaviour?

Here is the dockerfile I used to build the Rocker image:

```
FROM rocker/r-ver:4.2.1

WORKDIR /container

COPY container_test.r ./

CMD /bin/bash 
```

The container_test.r file contains only one line `cat("hello world")`. The image was build using:

`podman build -t testname/testcontainer:1.0 .`

And the Rocker-container instance was opened using:

`podman run --rm -ti -v .:/container testname/testcontainer:1.0`

I started an R 4.2.1 instance in the container, ran `install.packages("renv")` (likely also `library("renv")` missed in my notes) and then:

```
> init()

Welcome to renv!

It looks like this is your first time using renv. This is a one-time message,
briefly describing some of renv's functionality.

renv maintains a local cache of data on the filesystem, located at:

  - "~/.cache/R/renv"

This path can be customized: please see the documentation in `?renv::paths`.

renv will also write to files within the active project folder, including:

  - A folder 'renv' in the project directory, and
  - A lockfile called 'renv.lock' in the project directory.

In particular, projects using renv will normally use a private, per-project
R library, in which new packages will be installed. This project library is
isolated from other R libraries on your system.

In addition, renv will update files within your project directory, including:

  - .gitignore
  - .Rbuildignore
  - .Rprofile

Please read the introduction vignette with `vignette("renv")` for more information.
You can browse the package documentation online at https://rstudio.github.io/renv/.
Do you want to proceed? [y/N]:

>y
* "~/.cache/R/renv" has been created.
* Initializing project ...
cp -pPR '/usr/local/lib/R/site-library/renv/' '/container/renv/library/R-4.2/x86_64-pc-linux-gnu/.renv-copy-1635d4dd19'
================================================================================

cp: preserving times for '/container/renv/library/R-4.2/x86_64-pc-linux-gnu/.renv-copy-1635d4dd19/DESCRIPTION': Operation not permitted
cp: preserving times for '/container/renv/library/R-4.2/x86_64-pc-linux-gnu/.renv-copy-1635d4dd19/INDEX': Operation not permitted
cp: preserving times for '/container/renv/library/R-4.2/x86_64-pc-linux-gnu/.renv-copy-1635d4dd19/LICENSE': Operation not permitted
cp: preserving times for '/container/renv/library/R-4.2/x86_64-pc-linux-gnu/.renv-copy-1635d4dd19/Meta/Rd.rds': Operation not permitted
cp: preserving times for '/container/renv/library/R-4.2/x86_64-pc-linux-gnu/.renv-copy-1635d4dd19/Meta/features.rds': Operation not permitted
cp: preserving times for '/container/renv/library/R-4.2/x86_64-pc-linux-gnu/.renv-copy-1635d4dd19/Meta/hsearch.rds': Operation not permitted
cp: preserving times for '/container/renv/library/R-4.2/x86_64-pc-linux-gnu/.renv-copy-1635d4dd19/Meta/links.rds': Operation not permitted
cp: preserving times for '/container/renv/library/R-4.2/x86_64-pc-linux-gnu/.renv-copy-1635d4dd19/Meta/nsInfo.rds': Operation not permitted
cp: preserving times for '/container/renv/library/R-4.2/x86_64-pc-linux-gnu/.renv-copy-1635d4dd19/Meta/package.rds': Operation not permitted
cp: preserving times for '/container/renv/library/R-4.2/x86_64-pc-linux-gnu/.renv-copy-1635d4dd19/Meta/vignette.rds': Operation not permitted
cp: preserving times for '/container/renv/library/R-4.2/x86_64-pc-linux-gnu/.renv-copy-1635d4dd19/Meta': Operation not permitted
cp: preserving times for '/container/renv/library/R-4.2/x86_64-pc-linux-gnu/.renv-copy-1635d4dd19/NAMESPACE': Operation not permitted
cp: preserving times for '/container/renv/library/R-4.2/x86_64-pc-linux-gnu/.renv-copy-1635d4dd19/NEWS.md': Operation not permitted
cp: preserving times for '/container/renv/library/R-4.2/x86_64-pc-linux-gnu/.renv-copy-1635d4dd19/R/renv': Operation not permitted
cp: preserving times for '/container/renv/library/R-4.2/x86_64-pc-linux-gnu/.renv-copy-1635d4dd19/R/renv.rdb': Operation not permitted
cp: preserving times for '/container/renv/library/R-4.2/x86_64-pc-linux-gnu/.renv-copy-1635d4dd19/R/renv.rdx': Operation not permitted
cp: preserving times for '/container/renv/library/R-4.2/x86_64-pc-linux-gnu/.renv-copy-1635d4dd19/R': Operation not permitted
cp: preserving times for '/container/renv/library/R-4.2/x86_64-pc-linux-gnu/.renv-copy-1635d4dd19/bin/renv': Operation not permitted
cp: preserving times for '/container/renv/library/R-4.2/x86_64-pc-linux-gnu/.renv-copy-1635d4dd19/bin/renv.bat': Operation not permitted
cp: preserving times for '/container/renv/library/R-4.2/x86_64-pc-linux-gnu/.renv-copy-1635d4dd19/bin': Operation not permitted
cp: preserving times for '/container/renv/library/R-4.2/x86_64-pc-linux-gnu/.renv-copy-1635d4dd19/config.yml': Operation not permitted
cp: preserving times for '/container/renv/library/R-4.2/x86_64-pc-linux-gnu/.renv-copy-1635d4dd19/doc/bioconductor.R': Operation not permitted
cp: preserving times for '/container/renv/library/R-4.2/x86_64-pc-linux-gnu/.renv-copy-1635d4dd19/doc/bioconductor.Rmd': Operation not permitted
cp: preserving times for '/container/renv/library/R-4.2/x86_64-pc-linux-gnu/.renv-copy-1635d4dd19/doc/bioconductor.html': Operation not permitted
cp: preserving times for '/container/renv/library/R-4.2/x86_64-pc-linux-gnu/.renv-copy-1635d4dd19/doc/cellar.R': Operation not permitted
cp: preserving times for '/container/renv/library/R-4.2/x86_64-pc-linux-gnu/.renv-copy-1635d4dd19/doc/cellar.Rmd': Operation not permitted
cp: preserving times for '/container/renv/library/R-4.2/x86_64-pc-linux-gnu/.renv-copy-1635d4dd19/doc/cellar.html': Operation not permitted
cp: preserving times for '/container/renv/library/R-4.2/x86_64-pc-linux-gnu/.renv-copy-1635d4dd19/doc/ci.R': Operation not permitted
cp: preserving times for '/container/renv/library/R-4.2/x86_64-pc-linux-gnu/.renv-copy-1635d4dd19/doc/ci.Rmd': Operation not permitted
cp: preserving times for '/container/renv/library/R-4.2/x86_64-pc-linux-gnu/.renv-copy-1635d4dd19/doc/ci.html': Operation not permitted
cp: preserving times for '/container/renv/library/R-4.2/x86_64-pc-linux-gnu/.renv-copy-1635d4dd19/doc/collaborating.R': Operation not permitted
cp: preserving times for '/container/renv/library/R-4.2/x86_64-pc-linux-gnu/.renv-copy-1635d4dd19/doc/collaborating.Rmd': Operation not permitted
cp: preserving times for '/container/renv/library/R-4.2/x86_64-pc-linux-gnu/.renv-copy-1635d4dd19/doc/collaborating.html': Operation not permitted
cp: preserving times for '/container/renv/library/R-4.2/x86_64-pc-linux-gnu/.renv-copy-1635d4dd19/doc/docker.R': Operation not permitted
cp: preserving times for '/container/renv/library/R-4.2/x86_64-pc-linux-gnu/.renv-copy-1635d4dd19/doc/docker.Rmd': Operation not permitted
cp: preserving times for '/container/renv/library/R-4.2/x86_64-pc-linux-gnu/.renv-copy-1635d4dd19/doc/docker.html': Operation not permitted
cp: preserving times for '/container/renv/library/R-4.2/x86_64-pc-linux-gnu/.renv-copy-1635d4dd19/doc/faq.R': Operation not permitted
cp: preserving times for '/container/renv/library/R-4.2/x86_64-pc-linux-gnu/.renv-copy-1635d4dd19/doc/faq.Rmd': Operation not permitted
cp: preserving times for '/container/renv/library/R-4.2/x86_64-pc-linux-gnu/.renv-copy-1635d4dd19/doc/faq.html': Operation not permitted
cp: preserving times for '/container/renv/library/R-4.2/x86_64-pc-linux-gnu/.renv-copy-1635d4dd19/doc/index.html': Operation not permitted
cp: preserving times for '/container/renv/library/R-4.2/x86_64-pc-linux-gnu/.renv-copy-1635d4dd19/doc/lockfile.R': Operation not permitted
cp: preserving times for '/container/renv/library/R-4.2/x86_64-pc-linux-gnu/.renv-copy-1635d4dd19/doc/lockfile.Rmd': Operation not permitted
cp: preserving times for '/container/renv/library/R-4.2/x86_64-pc-linux-gnu/.renv-copy-1635d4dd19/doc/lockfile.html': Operation not permitted
cp: preserving times for '/container/renv/library/R-4.2/x86_64-pc-linux-gnu/.renv-copy-1635d4dd19/doc/mran.R': Operation not permitted
cp: preserving times for '/container/renv/library/R-4.2/x86_64-pc-linux-gnu/.renv-copy-1635d4dd19/doc/mran.Rmd': Operation not permitted
cp: preserving times for '/container/renv/library/R-4.2/x86_64-pc-linux-gnu/.renv-copy-1635d4dd19/doc/mran.html': Operation not permitted
cp: preserving times for '/container/renv/library/R-4.2/x86_64-pc-linux-gnu/.renv-copy-1635d4dd19/doc/packages.R': Operation not permitted
cp: preserving times for '/container/renv/library/R-4.2/x86_64-pc-linux-gnu/.renv-copy-1635d4dd19/doc/packages.Rmd': Operation not permitted
cp: preserving times for '/container/renv/library/R-4.2/x86_64-pc-linux-gnu/.renv-copy-1635d4dd19/doc/packages.html': Operation not permitted
cp: preserving times for '/container/renv/library/R-4.2/x86_64-pc-linux-gnu/.renv-copy-1635d4dd19/doc/profiles.R': Operation not permitted
cp: preserving times for '/container/renv/library/R-4.2/x86_64-pc-linux-gnu/.renv-copy-1635d4dd19/doc/profiles.Rmd': Operation not permitted
cp: preserving times for '/container/renv/library/R-4.2/x86_64-pc-linux-gnu/.renv-copy-1635d4dd19/doc/profiles.html': Operation not permitted
cp: preserving times for '/container/renv/library/R-4.2/x86_64-pc-linux-gnu/.renv-copy-1635d4dd19/doc/python.R': Operation not permitted
cp: preserving times for '/container/renv/library/R-4.2/x86_64-pc-linux-gnu/.renv-copy-1635d4dd19/doc/python.Rmd': Operation not permitted
cp: preserving times for '/container/renv/library/R-4.2/x86_64-pc-linux-gnu/.renv-copy-1635d4dd19/doc/python.html': Operation not permitted
cp: preserving times for '/container/renv/library/R-4.2/x86_64-pc-linux-gnu/.renv-copy-1635d4dd19/doc/renv.R': Operation not permitted
cp: preserving times for '/container/renv/library/R-4.2/x86_64-pc-linux-gnu/.renv-copy-1635d4dd19/doc/renv.Rmd': Operation not permitted
cp: preserving times for '/container/renv/library/R-4.2/x86_64-pc-linux-gnu/.renv-copy-1635d4dd19/doc/renv.html': Operation not permitted
cp: preserving times for '/container/renv/library/R-4.2/x86_64-pc-linux-gnu/.renv-copy-1635d4dd19/doc/rsconnect.R': Operation not permitted
cp: preserving times for '/container/renv/library/R-4.2/x86_64-pc-linux-gnu/.renv-copy-1635d4dd19/doc/rsconnect.Rmd': Operation not permitted
cp: preserving times for '/container/renv/library/R-4.2/x86_64-pc-linux-gnu/.renv-copy-1635d4dd19/doc/rsconnect.html': Operation not permitted
cp: preserving times for '/container/renv/library/R-4.2/x86_64-pc-linux-gnu/.renv-copy-1635d4dd19/doc/use.R': Operation not permitted
cp: preserving times for '/container/renv/library/R-4.2/x86_64-pc-linux-gnu/.renv-copy-1635d4dd19/doc/use.Rmd': Operation not permitted
cp: preserving times for '/container/renv/library/R-4.2/x86_64-pc-linux-gnu/.renv-copy-1635d4dd19/doc/use.html': Operation not permitted
cp: preserving times for '/container/renv/library/R-4.2/x86_64-pc-linux-gnu/.renv-copy-1635d4dd19/doc': Operation not permitted
cp: preserving times for '/container/renv/library/R-4.2/x86_64-pc-linux-gnu/.renv-copy-1635d4dd19/help/AnIndex': Operation not permitted
cp: preserving times for '/container/renv/library/R-4.2/x86_64-pc-linux-gnu/.renv-copy-1635d4dd19/help/aliases.rds': Operation not permitted
cp: preserving times for '/container/renv/library/R-4.2/x86_64-pc-linux-gnu/.renv-copy-1635d4dd19/help/figures/logo.svg': Operation not permitted
cp: preserving times for '/container/renv/library/R-4.2/x86_64-pc-linux-gnu/.renv-copy-1635d4dd19/help/figures': Operation not permitted
cp: preserving times for '/container/renv/library/R-4.2/x86_64-pc-linux-gnu/.renv-copy-1635d4dd19/help/paths.rds': Operation not permitted
cp: preserving times for '/container/renv/library/R-4.2/x86_64-pc-linux-gnu/.renv-copy-1635d4dd19/help/renv.rdb': Operation not permitted
cp: preserving times for '/container/renv/library/R-4.2/x86_64-pc-linux-gnu/.renv-copy-1635d4dd19/help/renv.rdx': Operation not permitted
cp: preserving times for '/container/renv/library/R-4.2/x86_64-pc-linux-gnu/.renv-copy-1635d4dd19/help': Operation not permitted
cp: preserving times for '/container/renv/library/R-4.2/x86_64-pc-linux-gnu/.renv-copy-1635d4dd19/html/00Index.html': Operation not permitted
cp: preserving times for '/container/renv/library/R-4.2/x86_64-pc-linux-gnu/.renv-copy-1635d4dd19/html/R.css': Operation not permitted
cp: preserving times for '/container/renv/library/R-4.2/x86_64-pc-linux-gnu/.renv-copy-1635d4dd19/html': Operation not permitted
cp: preserving times for '/container/renv/library/R-4.2/x86_64-pc-linux-gnu/.renv-copy-1635d4dd19/repos/src/contrib/PACKAGES': Operation not permitted
cp: preserving times for '/container/renv/library/R-4.2/x86_64-pc-linux-gnu/.renv-copy-1635d4dd19/repos/src/contrib/PACKAGES.gz': Operation not permitted
cp: preserving times for '/container/renv/library/R-4.2/x86_64-pc-linux-gnu/.renv-copy-1635d4dd19/repos/src/contrib/PACKAGES.rds': Operation not permitted
cp: preserving times for '/container/renv/library/R-4.2/x86_64-pc-linux-gnu/.renv-copy-1635d4dd19/repos/src/contrib/renv_0.16.0.tar.gz': Operation not permitted
cp: preserving times for '/container/renv/library/R-4.2/x86_64-pc-linux-gnu/.renv-copy-1635d4dd19/repos/src/contrib': Operation not permitted
cp: preserving times for '/container/renv/library/R-4.2/x86_64-pc-linux-gnu/.renv-copy-1635d4dd19/repos/src': Operation not permitted
cp: preserving times for '/container/renv/library/R-4.2/x86_64-pc-linux-gnu/.renv-copy-1635d4dd19/repos': Operation not permitted
cp: preserving times for '/container/renv/library/R-4.2/x86_64-pc-linux-gnu/.renv-copy-1635d4dd19/resources/WELCOME': Operation not permitted
cp: preserving times for '/container/renv/library/R-4.2/x86_64-pc-linux-gnu/.renv-copy-1635d4dd19/resources/activate.R': Operation not permitted
cp: preserving times for '/container/renv/library/R-4.2/x86_64-pc-linux-gnu/.renv-copy-1635d4dd19/resources/scripts-git-askpass.cmd': Operation not permitted
cp: preserving times for '/container/renv/library/R-4.2/x86_64-pc-linux-gnu/.renv-copy-1635d4dd19/resources/scripts-git-askpass.sh': Operation not permitted
cp: preserving times for '/container/renv/library/R-4.2/x86_64-pc-linux-gnu/.renv-copy-1635d4dd19/resources/scripts-repos-cache.R': Operation not permitted
cp: preserving times for '/container/renv/library/R-4.2/x86_64-pc-linux-gnu/.renv-copy-1635d4dd19/resources': Operation not permitted
cp: preserving times for '/container/renv/library/R-4.2/x86_64-pc-linux-gnu/.renv-copy-1635d4dd19/rstudio/addins.dcf': Operation not permitted
cp: preserving times for '/container/renv/library/R-4.2/x86_64-pc-linux-gnu/.renv-copy-1635d4dd19/rstudio': Operation not permitted
cp: preserving times for '/container/renv/library/R-4.2/x86_64-pc-linux-gnu/.renv-copy-1635d4dd19': 
Operation not permitted
Error: error copying directory [error code 1]
Traceback (most recent calls last):
7: init()
6: renv_imbue_impl(project)
5: renv_imbue_self(project)
4: renv_imbue_self_binary(source, target)
3: renv_file_copy(source, target, overwrite = TRUE)
2: renv_file_copy_dir(source, target)
1: stop(status)
```

Because the previous run was problematic, I also tried `init(bare = TRUE)` with similar results. Then, when I tried to build another Rocker image (which shouldn't have nothing to do with the renv) the Podman failed:

```
podman build -t testname/testcontainer:1.0 .
time="2023-05-24T12:18:29+03:00" level=error msg="1 error occurred:\n\t* open C:\\some_directories\\renv\\sandbox\\R-4.2\\x86_64-pc-linux-gnu\\25ebdc09\\KernSmooth: The file cannot be accessed by the system.\n\n\n"
Error: Post "http://d/v4.5.0/libpod/build?dockerfile=%5B%22Dockerfile%22%5D&forcerm=1&httpproxy=1&identitylabel=1&idmappingoptions=%7B%22HostUIDMapping%22%3Atrue%2C%22HostGIDMapping%22%3Atrue%2C%22UIDMap%22%3A%5B%5D%2C%22GIDMap%22%3A%5B%5D%2C%22AutoUserNs%22%3Afalse%2C%22AutoUserNsOpts%22%3A%7B%22Size%22%3A0%2C%22InitialSize%22%3A0%2C%22PasswdFile%22%3A%22%22%2C%22GroupFile%22%3A%22%22%2C%22AdditionalUIDMappings%22%3Anull%2C%22AdditionalGIDMappings%22%3Anull%7D%7D&isolation=1&jobs=1&layers=1&networkmode=0&nsoptions=%5B%7B%22Name%22%3A%22user%22%2C%22Host%22%3Atrue%2C%22Path%22%3A%22%22%7D%5D&omithistory=0&output=testname%2Ftestcontainer%3A1.0&outputformat=application%2Fvnd.oci.image.manifest.v1%2Bjson&platform=%2F&pullpolicy=missing&rm=1&shmsize=67108864&t=testname%2Ftestcontainer%3A1.0": 
io: read/write on closed pipe
```

I believed that experimenting with the renv in a rootless Rocker container would be safe, but something obviously went wrong. As carefully as I can, I removed and reinstalled both the Podman and WSL following these recommendations, but the issue didn't go away: https://superuser.com/questions/1619233/completely-reinstall-wsl/1755410#1755410

The podman build does not work even if I change the `FROM rocker/r-ver:4.2.1` to `FROM rocker/r-ver:4.3.0` in the dockerfile, the podman still tries to access the earlier version: `… renv\\sandbox\\R-4.2\\x86_64-pc-linux-gnu\\25ebdc09\\KernSmooth`. However, I am able to build the image of this totally unrelated tutorial: https://devopscube.com/podman-tutorial-beginners/

Here are some insights from the renv experts. https://github.com/rstudio/renv/issues/1335
 
This is getting quite complicated, could somebody shed some light on what could be the issue?

### Steps to reproduce the issue

Rough steps to reproduce the issue:

1. On Windows 10 Enterprise with WSL2 installed, build the image and start a rocker/r-ver:4.2.1-container instance with Podman 4.5.0.

2. In the container, start R 4.2.1 instance and run `install.packages("renv")`. I am quite sure that `library("renv")` was also used but missed in my notes.

3. Run `init()`, possibly also `init(bare = TRUE)`.

4. Exit the container and try to rebuild the Rocker image which should now fail.

### Describe the results you received

Rebuilding the Rocker images with the Podman failed although the previous container instance was shutdown. And the problem persists even if the Podman and WSL were removed and reinstalled.

### Describe the results you expected

I expected that experimenting within an R instance contained by a rootless Rocker container is safe and shouldn't effect on the coming image builds.

### podman info output

```
podman info
host:
  arch: amd64
  buildahVersion: 1.30.0
  cgroupControllers: []
  cgroupManager: cgroupfs
  cgroupVersion: v1
  conmon:
    package: conmon-2.1.7-2.fc37.x86_64
    path: /usr/bin/conmon
    version: 'conmon version 2.1.7, commit: '
  cpuUtilization:
    idlePercent: 99.6
    systemPercent: 0.25
    userPercent: 0.15
  cpus: 8
  databaseBackend: boltdb
  distribution:
    distribution: fedora
    variant: container
    version: "37"
  eventLogger: journald
  hostname: WKS-88907-LT
  idMappings:
    gidmap:
    - container_id: 0
      host_id: 1000
      size: 1
    - container_id: 1
      host_id: 524288
      size: 65536
    uidmap:
    - container_id: 0
      host_id: 1000
      size: 1
    - container_id: 1
      host_id: 524288
      size: 65536
  kernel: 5.15.90.1-microsoft-standard-WSL2
  linkmode: dynamic
  logDriver: journald
  memFree: 7661056000
  memTotal: 8264810496
  networkBackend: netavark
  ociRuntime:
    name: crun
    package: crun-1.8.5-1.fc37.x86_64
    path: /usr/bin/crun
    version: |-
      crun version 1.8.5
      commit: b6f80f766c9a89eb7b1440c0a70ab287434b17ed
      rundir: /run/user/1000/crun
      spec: 1.0.0
      +SYSTEMD +SELINUX +APPARMOR +CAP +SECCOMP +EBPF +CRIU +LIBKRUN +WASM:wasmedge +YAJL
  os: linux
  remoteSocket:
    exists: true
    path: /run/user/1000/podman/podman.sock
  security:
    apparmorEnabled: false
    capabilities: CAP_CHOWN,CAP_DAC_OVERRIDE,CAP_FOWNER,CAP_FSETID,CAP_KILL,CAP_NET_BIND_SERVICE,CAP_SETFCAP,CAP_SETGID,CAP_SETPCAP,CAP_SETUID,CAP_SYS_CHROOT
    rootless: true
    seccompEnabled: true
    seccompProfilePath: /usr/share/containers/seccomp.json
    selinuxEnabled: false
  serviceIsRemote: true
  slirp4netns:
    executable: /usr/bin/slirp4netns
    package: slirp4netns-1.2.0-8.fc37.x86_64
    version: |-
      slirp4netns version 1.2.0
      commit: 656041d45cfca7a4176f6b7eed9e4fe6c11e8383
      libslirp: 4.7.0
      SLIRP_CONFIG_VERSION_MAX: 4
      libseccomp: 2.5.3
  swapFree: 2147483648
  swapTotal: 2147483648
  uptime: 0h 7m 52.00s
plugins:
  authorization: null
  log:
  - k8s-file
  - none
  - passthrough
  - journald
  network:
  - bridge
  - macvlan
  - ipvlan
  volume:
  - local
registries:
  search:
  - docker.io
store:
  configFile: /home/user/.config/containers/storage.conf
  containerStore:
    number: 0
    paused: 0
    running: 0
    stopped: 0
  graphDriverName: overlay
  graphOptions: {}
  graphRoot: /home/user/.local/share/containers/storage
  graphRootAllocated: 1081101176832
  graphRootUsed: 648327168
  graphStatus:
    Backing Filesystem: extfs
    Native Overlay Diff: "true"
    Supports d_type: "true"
    Using metacopy: "false"
  imageCopyTmpDir: /var/tmp
  imageStore:
    number: 0
  runRoot: /run/user/1000/containers
  transientStore: false
  volumePath: /home/user/.local/share/containers/storage/volumes
version:
  APIVersion: 4.5.0
  Built: 1681486976
  BuiltTime: Fri Apr 14 18:42:56 2023
  GitCommit: ""
  GoVersion: go1.19.7
  Os: linux
  OsArch: linux/amd64
  Version: 4.5.0
```

### Podman in a container

No

### Privileged Or Rootless

Rootless

### Upstream Latest Release

No

**Comment by vrothberg** [comment]
Thanks for reaching out!

> podman run --rm -ti -v .:/container testname/testcontainer:1.0

I assume the volume mount is causing issue, especially given you get EPERMs on the mounted paths.  Mounting host paths into a rootless container has some peculiarities.  I recommend reading the following article and try the recommended solutions https://www.redhat.com/sysadmin/debug-rootless-podman-mounted-volumes.

**Comment by IlmariTamminen** [comment]
> Thanks for reaching out!
> 
> > podman run --rm -ti -v .:/container testname/testcontainer:1.0
> 
> I assume the volume mount is causing issue, especially given you get EPERMs on the mounted paths. Mounting host paths into a rootless container has some peculiarities. I recommend reading the following article and try the recommended solutions https://www.redhat.com/sysadmin/debug-rootless-podman-mounted-volumes.

Thank you. Before getting familiar with the material, I would like to verify is the assumption justified? I mean it sounds odd that mounting during the `podman run` (usually done afterwards) can affect on the `podman build`, which are separate steps when dealing with the images and containers. The main problem was with the `podman build`.

**Comment by vrothberg** [comment]
Thanks, @IlmariTamminen.  I was too hasty.  The build issue you see is the following, right?

> podman build -t testname/testcontainer:1.0 .
time="2023-05-24T12:18:29+03:00" level=error msg="1 error occurred:\n\t* open C:\\some_directories\\renv\\sandbox\\R-4.2\\x86_64-pc-linux-gnu\\25ebdc09\\KernSmooth: The file cannot be accessed by the system.\n\n\n"
Error: Post "http://d/v4.5.0/libpod/build?dockerfile=%5B%22Dockerfile%22%5D&forcerm=1&httpproxy=1&identitylabel=1&idmappingoptions=%7B%22HostUIDMapping%22%3Atrue%2C%22HostGIDMapping%22%3Atrue%2C%22UIDMap%22%3A%5B%5D%2C%22GIDMap%22%3A%5B%5D%2C%22AutoUserNs%22%3Afalse%2C%22AutoUserNsOpts%22%3A%7B%22Size%22%3A0%2C%22InitialSize%22%3A0%2C%22PasswdFile%22%3A%22%22%2C%22GroupFile%22%3A%22%22%2C%22AdditionalUIDMappings%22%3Anull%2C%22AdditionalGIDMappings%22%3Anull%7D%7D&isolation=1&jobs=1&layers=1&networkmode=0&nsoptions=%5B%7B%22Name%22%3A%22user%22%2C%22Host%22%3Atrue%2C%22Path%22%3A%22%22%7D%5D&omithistory=0&output=testname%2Ftestcontainer%3A1.0&outputformat=application%2Fvnd.oci.image.manifest.v1%2Bjson&platform=%2F&pullpolicy=missing&rm=1&shmsize=67108864&t=testname%2Ftestcontainer%3A1.0": 
io: read/write on closed pipe

@flouthoc PTAL

**Comment by IlmariTamminen** [comment]
No problem. Yes, the stuck image build is the problem.

**Comment by IlmariTamminen** [comment]
@vrothberg Anything new regarding the issue? The issue was left closed, I was just checking that this does not get forgotten.

**Comment by flouthoc** [comment]
Sorry I missed checking this, let me re-open this.

**Comment by IlmariTamminen** [comment]
Thanks @flouthoc. Okay, the problem currently is that I have changed my computer, and "the issue is now gone with the earlier one". I also discarded the renv R package anyhow. My primary motivation was to find out have I done something silly or was there important flaws with the Podman I should be aware of in the future. If I face similar problems again, I can try your tip. It seems that the issue wasn't serious.
